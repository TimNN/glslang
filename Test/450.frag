#version 450 core

in float in1;
in vec2 in2;
in vec3 in3;
in vec4 in4;

void main()
{
    vec2 v2 = dFdxFine(in2);
    vec3 v3 = dFdyCoarse(in3);
    vec4 v4 = fwidth(in4);
    v4 = dFdyFine(in4);
    v3 = dFdyFine(in3);
    float f = dFdx(in1) + dFdxFine(in1) + dFdxCoarse(in1);
    v4 = fwidthCoarse(in4) + fwidthFine(in4);

    float cull = gl_CullDistance[2];
    float consts = gl_MaxCullDistances + gl_MaxCombinedClipAndCullDistances;
}
