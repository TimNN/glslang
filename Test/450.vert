#version 450 core

out gl_PerVertex {
    float gl_CullDistance[3];
};

void main()
{
    gl_CullDistance[2] = 4.5;
}
