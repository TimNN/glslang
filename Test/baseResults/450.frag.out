450.frag
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.

Shader version: 450
0:? Sequence
0:8  Function Definition: main( (void)
0:8    Function Parameters: 
0:10    Sequence
0:10      Sequence
0:10        move second child to first child (2-component vector of float)
0:10          'v2' (2-component vector of float)
0:10          dPdxFine (2-component vector of float)
0:10            'in2' (smooth in 2-component vector of float)
0:11      Sequence
0:11        move second child to first child (3-component vector of float)
0:11          'v3' (3-component vector of float)
0:11          dPdyCoarse (3-component vector of float)
0:11            'in3' (smooth in 3-component vector of float)
0:12      Sequence
0:12        move second child to first child (4-component vector of float)
0:12          'v4' (4-component vector of float)
0:12          fwidth (4-component vector of float)
0:12            'in4' (smooth in 4-component vector of float)
0:13      move second child to first child (4-component vector of float)
0:13        'v4' (4-component vector of float)
0:13        dPdyFine (4-component vector of float)
0:13          'in4' (smooth in 4-component vector of float)
0:14      move second child to first child (3-component vector of float)
0:14        'v3' (3-component vector of float)
0:14        dPdyFine (3-component vector of float)
0:14          'in3' (smooth in 3-component vector of float)
0:15      Sequence
0:15        move second child to first child (float)
0:15          'f' (float)
0:15          add (float)
0:15            add (float)
0:15              dPdx (float)
0:15                'in1' (smooth in float)
0:15              dPdxFine (float)
0:15                'in1' (smooth in float)
0:15            dPdxCoarse (float)
0:15              'in1' (smooth in float)
0:16      move second child to first child (4-component vector of float)
0:16        'v4' (4-component vector of float)
0:16        add (4-component vector of float)
0:16          fwidthCoarse (4-component vector of float)
0:16            'in4' (smooth in 4-component vector of float)
0:16          fwidthFine (4-component vector of float)
0:16            'in4' (smooth in 4-component vector of float)
0:18      Sequence
0:18        move second child to first child (float)
0:18          'cull' (float)
0:18          direct index (smooth float)
0:18            'gl_CullDistance' (smooth in implicitly-sized array of float)
0:18            Constant:
0:18              2 (const int)
0:19      Sequence
0:19        move second child to first child (float)
0:19          'consts' (float)
0:19          Constant:
0:19            16.000000
0:?   Linker Objects
0:?     'in1' (smooth in float)
0:?     'in2' (smooth in 2-component vector of float)
0:?     'in3' (smooth in 3-component vector of float)
0:?     'in4' (smooth in 4-component vector of float)
0:?     'gl_CullDistance' (smooth in implicitly-sized array of float)


Linked fragment stage:


Shader version: 450
0:? Sequence
0:8  Function Definition: main( (void)
0:8    Function Parameters: 
0:10    Sequence
0:10      Sequence
0:10        move second child to first child (2-component vector of float)
0:10          'v2' (2-component vector of float)
0:10          dPdxFine (2-component vector of float)
0:10            'in2' (smooth in 2-component vector of float)
0:11      Sequence
0:11        move second child to first child (3-component vector of float)
0:11          'v3' (3-component vector of float)
0:11          dPdyCoarse (3-component vector of float)
0:11            'in3' (smooth in 3-component vector of float)
0:12      Sequence
0:12        move second child to first child (4-component vector of float)
0:12          'v4' (4-component vector of float)
0:12          fwidth (4-component vector of float)
0:12            'in4' (smooth in 4-component vector of float)
0:13      move second child to first child (4-component vector of float)
0:13        'v4' (4-component vector of float)
0:13        dPdyFine (4-component vector of float)
0:13          'in4' (smooth in 4-component vector of float)
0:14      move second child to first child (3-component vector of float)
0:14        'v3' (3-component vector of float)
0:14        dPdyFine (3-component vector of float)
0:14          'in3' (smooth in 3-component vector of float)
0:15      Sequence
0:15        move second child to first child (float)
0:15          'f' (float)
0:15          add (float)
0:15            add (float)
0:15              dPdx (float)
0:15                'in1' (smooth in float)
0:15              dPdxFine (float)
0:15                'in1' (smooth in float)
0:15            dPdxCoarse (float)
0:15              'in1' (smooth in float)
0:16      move second child to first child (4-component vector of float)
0:16        'v4' (4-component vector of float)
0:16        add (4-component vector of float)
0:16          fwidthCoarse (4-component vector of float)
0:16            'in4' (smooth in 4-component vector of float)
0:16          fwidthFine (4-component vector of float)
0:16            'in4' (smooth in 4-component vector of float)
0:18      Sequence
0:18        move second child to first child (float)
0:18          'cull' (float)
0:18          direct index (smooth float)
0:18            'gl_CullDistance' (smooth in 3-element array of float)
0:18            Constant:
0:18              2 (const int)
0:19      Sequence
0:19        move second child to first child (float)
0:19          'consts' (float)
0:19          Constant:
0:19            16.000000
0:?   Linker Objects
0:?     'in1' (smooth in float)
0:?     'in2' (smooth in 2-component vector of float)
0:?     'in3' (smooth in 3-component vector of float)
0:?     'in4' (smooth in 4-component vector of float)
0:?     'gl_CullDistance' (smooth in 3-element array of float)

